### ISSUE TITLE
[BUG]: `test_data` getting used for training

### ISSUE THREAD

**Issue by ngupta23** [pycaret/pycaret#2528]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the develop branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@develop).

### Issue Description

When providing the `test_data` argument, the data in this dataframe is getting added to the training data and being used for training. This should not happen. `test_data` should only be used as a holdout dataset.

### Reproducible Example

```python
!pip install git+https://github.com/pycaret/pycaret.git@develop

import pandas as pd
from sktime.datasets import load_arrow_head

# Load all arrow head
arrow_train_X, arrow_train_y = load_arrow_head(split="train", return_type="numpy2d")
arrow_test_X, arrow_test_y = load_arrow_head(split="test", return_type="numpyflat")

train_X_pd = pd.DataFrame(arrow_train_X)
train_y_pd = pd.DataFrame({"target": arrow_train_y})
test_X_pd = pd.DataFrame(arrow_test_X)
test_y_pd = pd.DataFrame({"target": arrow_test_y})

train_data = pd.concat([train_X_pd, train_y_pd], axis=1)
test_data = pd.concat([test_X_pd, test_y_pd], axis=1)

# Needed due to another bug in pycaret where test data columns are not coerced to string but train columns are
train_data.columns = [str(col) for col in train_data.columns]
test_data.columns = [str(col) for col in test_data.columns]

print(train_data.shape, test_data.shape)

from pycaret.classification import ClassificationExperiment
exp1 = ClassificationExperiment()
exp1.setup(data = train_data, test_data=test_data, target="target")
print(exp1.X_train.shape, exp1.X_test.shape)
```

### Expected Behavior

Expected behavior is to get 36 samples in the train dataset and 175 in the test dataset. But pycaret is adding all of then in the train dataset. This should not happen

![image](https://user-images.githubusercontent.com/33585645/167723733-5e1ed92f-c8d0-4c19-b281-dd4dda8c290a.png)

![image](https://user-images.githubusercontent.com/33585645/167723982-b8272f64-e5af-4a6d-9f11-7b62487a8bbd.png)

### Actual Results

```python-traceback
See above.
```

### Installed Versions

!pip install git+https://github.com/pycaret/pycaret.git@develop

**Comment by tvdboom** [comment]
Thanks for noticing @ngupta23! I fixed in #2530. Please review it.

**Comment by ngupta23** [comment]
I verified and it works fine now. Thanks!
